### PR TITLE
Updated PromptComponent to support checkbox

### PR DESCRIPTION
**Pull request by Yashsharma1911**
**Notes for Reviewers**

This PR fixes #

This PR update PromptComponent to support checkbox with text `Do not show again`
```diff
const response = await resetRef.current.show({
    title: "Reset Canvas?",
    subtitle,
    options: ["RESET", "CANCEL"],
+   showCheckbox: true,
});
```

And to access checkbox state, you can call `getCheckboxState` from ref

```
const checkboxState = resetRef.current.getCheckboxState();
```

Screenshot 2023-09-05 at 6 23 16 PM

```

```

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by Yashsharma1911** [comment]
`Todo:` A cross in modal to make it consistent

File: ui/components/PromptComponent.js [hunk 1]
```diff
@@ -116,6 +132,20 @@ class PromptComponent extends React.Component {
                   {subtitle}
                 </Typography>
               </DialogContentText>
+              {showCheckbox && (
+                <FormControlLabel
+                  control={
+                    <Checkbox
+                      checked={isChecked}
+                      onChange={() => {
+                        this.handleCheckboxChange();
```
Comment (Ckbhatia) [inline ui/components/PromptComponent.js:141]: This extra function isn't required

File: ui/components/PromptComponent.js [hunk 2]
```diff
@@ -89,9 +94,20 @@ class PromptComponent extends React.Component {
     this.setState({ show : false });
   };
 
+  handleCheckboxChange = () => {
+    this.setState((prevState) => ({
+      isChecked : !prevState.isChecked,
+    }));
+  };
+
+  getCheckboxState = () => {
```
Comment (Ckbhatia) [inline ui/components/PromptComponent.js:103]: What is the use case of this function?

File: ui/components/PromptComponent.js [hunk 3]
```diff
@@ -80,6 +84,7 @@ class PromptComponent extends React.Component {
         title : passed.title,
         subtitle : passed.subtitle,
         options : passed.options,
+        showCheckbox : passed.showCheckbox === true,
```
Comment (Ckbhatia) [inline ui/components/PromptComponent.js:87]: You can do this instead
```suggestion
        showCheckbox : !!passed.showCheckbox,
```

**Review by Ckbhatia — CHANGES_REQUESTED** [review]

File: ui/components/PromptComponent.js (hunk 2 above)
Comment (Yashsharma1911) [reply ui/components/PromptComponent.js:103]: This method function is required to get state of checkbox through ref, although I'm open for suggestion if you have

File: ui/components/PromptComponent.js (hunk 1 above)
Comment (Yashsharma1911) [reply ui/components/PromptComponent.js:141]: I created this function to make things clean, although we can update state directly here, also Is that a good practice as clean code?

**Comment by leecalcote** [comment]
What the styling of the checkbox changed from how it looks in the screenshot?

File: ui/components/PromptComponent.js (hunk 1 above)
Comment (Ckbhatia) [reply ui/components/PromptComponent.js:141]:
```
onChange={this.handleCheckboxChange}
```
Isn't simple?

Comment (Yashsharma1911) [reply ui/components/PromptComponent.js:141]: Yes it is thanks for that last time actually I was doing multiple things on that so I added callback function and forgot to remove, now I get what you are talking about, it make sense

**Comment by Yashsharma1911** [comment]
Screenshot 2023-09-11 at 4 36 49 PM

**Comment by Yashsharma1911** [comment]
I'm merging this PR

**Pull request merged**